### ISSUE TITLE
please consider adding the Infinality patches

### ISSUE THREAD

**Issue by RJVB** [KaOSx/core#18]
Please consider adding the Infinality patches to the freetype2 package as well as the corresponding patches to fontconfig and cairo. This is a (feature) request but in practice the gain in font rendering quality is so large that I consider it almost a bug to omit the patches in question. See the screenshot which shows a Konsole window before (left) and after (right) installing the freetype2-infinality package.

Up-to-date patches can be found in archfan's repository, a fork of bohoomil's invaluable work on the legacy Infinality patchset made compatible with Freetype 2.7+:
https://github.com/archfan/infinality_bundle

Note also the issue with glibc 2.25+ that probably affects building the standard fontconfig too:
https://github.com/archfan/infinality_bundle/issues/9

![screenshot_20170303_165411](https://cloud.githubusercontent.com/assets/1756884/23559331/ed5911e2-0036-11e7-846c-9207ac49efc2.png)

**Comment by demmm** [comment]
It has been considered a few times to patch, but too many times there are build issues with other packages on updates, so at this stage the cons of breaking others has not outweighed the pros of going with patched freetype2.
In general, KaOS avoids patching upstream.  Patches are basically only used to fix breakage or a CVE that has no new release yet, not to customize.  Reasoning if it really is better, why is upstream not accepting the patches?

**Comment by RJVB** [comment]
Hmmm, I've been using these patches (and the accompanying ones to FontConfig and Cairo) for years now, on both Linux and Mac, without ever a problem. They don't change the API or even ABI. There can be lags if FreeType is updated and changed too much for the patches to catch up immediately but then there's always the choice between staying on the previous version or running without the patches for a while. On KaOS I did the latter after FreeType hit 2.7 because the Cairo binary package used a new function not available in Freetype 2.6 . That episode made me appreciate the additional quality even more.

Does KaOS know something like alternative (or virtual) packages, where a dependency can be satisfied by more than 1 package,or else install variants? I wouldn't mind instructing the system I want the +infinality variant of Freetype, FontConfig and Cairo, and then putting up with those packages building from source during upgrades.

Why upstream never accepted the full patchset is indeed a good question. A large part of the Infinality patches *were* accepted at some point in the past. From what I know there were/are 2 considerations at play:
- someone upstream feared there were licensing issues in the part not included, without ground from what I understand
- someone upstream considered the remaining changes too expensive, made rendering too slow. There are still comments in the code and documentation hinting at that, but without any quantification. It seems likely that better rendering indeed comes at a cost (if not only because more pixels are being generated and drawn). I would guess that cost is significant only in rather artificial benchmarks and gets drowned in the noise as soon as actual rendering starts taking place. I have used the patches on a now 5yo netbook that was already really slow when new (AMD C60 CPU) and even there I never noticed any difference.

To be honest I wouldn't be very surprised if there were other factors at play here. With those remaining patches and a patched FontConfig text quality becomes probably even better at regular screen display sizes (and resolutions) than it can be with ClearType and IMHO certainly so than with Apple's CoreText, and someone might have incentives to prevent that in the official version.
